rust/feed/src/update/mod.rs

```diff
@@ -154,30 +166,46 @@ where
         }
         Err(ErrorKind::MissingExit(key.value()))
     }
+
     /// Perform a signature check of the sha256sums file
     pub fn verify_signature(&self) -> Result<(), verify::Error> {
-        //self::SignatureChecker::signature_check(&path)
         let path = self.loader.root_path().unwrap();
         crate::verify::check_signature(&path)
     }
-}
 
-impl<'a, S, L, V, R> Iterator for Update<'a, S, L, V>
-where
-    S: Sync + Send + Dispatcher,
-    L: Sync + Send + Loader + AsBufReader<File>,
-    V: Iterator<Item = Result<HashSumFileItem<'a, R>, verify::Error>>,
-    R: Read + 'a,
-{
-    type Item = Result<String, Error>;
+    /// Run the feed update and log each result with the
+    /// given log level. If an error occurs, return it.
+    pub async fn perform_update(self, log_level: Level) -> Result<(), Error> {
+        let results = self.stream().collect::<Vec<_>>().await;
+        for result in results.iter() {
+            match log_level {
+                Level::TRACE => trace!(?result),
```

Comment:
what's the reasoning for adding a log level into this function? I'd guess you only need the result of it in tracing?

Reply:
I was trying to keep the same behavior as before and in scannerctl we called this with `WARN` log level. Would happily remove this though.

Reply:
please do, the WARN log level was by accident.

Reply:
Done